flip-shop_app/src/pages/products/Product.jsx

```diff
@@ -1,55 +1,49 @@
 import "./product-filter.css";
-// import { products } from "../../data";
-import { productImg1, productImg2 } from "../../assets/images/photos";
+import axios from 'axios';
+import React, { useEffect, useState } from "react";
 import { ProductsList } from "./components/product-all";
 import { Productfilter } from "./components/product-filter";
+
 export default function ProductWithFilter() {
+  const [productData, setProductData] = useState([]);
+  useEffect(() => {
+    (async function () {
+      try {
+        await axios.get("/api/products").then((res) => {
+          setProductData(res.data.products);
+        });
+      } catch (error) {
+        console.error(error);
+        setProductData("failed do fetch data", error);
```

Comment:
If there is an error, don't set products with an error, just display that error.

Reply:
okay. i will update it.